src/Core/Utilities/RequireFeatureAttribute.cs

```diff
@@ -30,7 +31,7 @@ public override void OnActionExecuting(ActionExecutingContext context)
 
         if (!featureService.IsEnabled(_featureFlagKey, currentContext))
         {
-            throw new FeatureUnavailableException();
+            context.Result = new NotFoundObjectResult("This feature is unavailable.");
```

Comment:
❓ If this setting of the result is acceptable to our middleware and everything works as expected we'd be complete with this change here, but per previous comments it isn't in our situation right? If we can make it move that way that's great, but I thought we need an exception to throw.

Reply:
I've tested this and confirm that it returns a 404 error with the message. Also note [per the docs](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.filters.actionexecutingcontext.result?view=aspnetcore-7.0&viewFallbackFrom=net-6.0):

> Setting Result to a non-null value inside an action filter will short-circuit the action and any remaining action filters.

Reply:
If we mix patterns but this is the recommended way and we'd eventually want to move everything to it, I support the mixing.